hi-ml/testhiml/testhiml/test_regression_tests.py

```diff
@@ -0,0 +1,239 @@
+    data1 = bytes([1, 2, 3])
+    data2 = bytes([4, 5, 6])
+    expected.write_bytes(data1)
+    actual.write_bytes(data1)
+    assert compare_files(expected=expected, actual=actual) == ""
+    actual.write_bytes(data2)
+    assert compare_files(expected=expected, actual=actual) == CONTENTS_MISMATCH
+
+
+def test_compare_folder(tmp_path: Path) -> None:
+    """
+    Test the comparison of folders that we use for regression tests.
+    """
+    # Create a test of expected and actual files on the fly.
+    expected = tmp_path / "expected"
+    actual = tmp_path / "actual"
+    matching = "matching.txt"
+    missing = "missing.txt"
+    ignored = "ignored.txt"
+    # Comparison should cover at least .csv and .txt files
+    mismatch = "mismatch.csv"
+    extra = "extra.txt"
+    subfolder = Path("folder")
+    # This file exists in both expected and actual, should not raise any alerts because it contents matches
+    # apart from linebreaks
+    create_folder_and_write_text(expected / subfolder / matching, "Line1\r\nLine2")
+    create_folder_and_write_text(actual / subfolder / matching, "Line1\nLine2")
+    # This file only exists in the expected results, and should create an error
+    (expected / subfolder / missing).write_text("missing")
+    (actual / extra).write_text("extra")
+    # This file exists in both actual and expected, but has different contents, hence should create an error
+    (expected / subfolder / mismatch).write_text("contents1")
+    (actual / subfolder / mismatch).write_text("contents2")
+
+    messages = compare_folder_contents(expected_folder=expected, actual_folder=actual, csv_relative_tolerance=0.0)
+    all_messages = " ".join(messages)
+    # No issues expected
+    assert matching not in all_messages
+    assert extra not in all_messages
+    assert ignored not in all_messages
+    # Folders should be skipped in the comparison
+    assert f"{MISSING_FILE}: {subfolder}" not in messages
+    assert f"{MISSING_FILE}: {subfolder}/{missing}" in messages
+    assert f"{CONTENTS_MISMATCH}: {subfolder}/{mismatch}" in messages
+
+
+def test_compare_plain_outputs(tmp_path: Path) -> None:
+    """
+    Test if we can compare that a set of files from the job outputs.
+    """
+    expected = tmp_path / REGRESSION_TEST_OUTPUT_FOLDER
+    actual = tmp_path / "my_output"
+    for folder in [expected, actual]:
+        file1 = folder / "output.txt"
+        create_folder_and_write_text(file1, "Something")
+    # First comparison should pass
+    compare_folders_and_run_outputs(expected=expected, actual=actual, csv_relative_tolerance=0.0)
+    # Now add a file to the set of expected files that does not exist in the run: comparison should now fail
+    no_such_file = "no_such_file.txt"
+    file2 = expected / no_such_file
+    create_folder_and_write_text(file2, "foo")
+    with pytest.raises(ValueError) as ex:
+        compare_folders_and_run_outputs(expected=tmp_path, actual=Path.cwd(), csv_relative_tolerance=0.0)

```

Comment:
Is there a reason we compare tmp_path and cwd here rather than expected and actual? Isn't it possible that tmp_path and cwd already did not match?

Reply:
good catch. fixed.